index.js

```diff
@@ -11,11 +11,21 @@ function toObject(val) {
 	return Object(val);
 }
 
-function isFunc(val) {
-	return typeof val === 'function';
+function assignKey(to, from, key) {
+	var val = from[key];
+
+	if (typeof val === 'undefined' || val === null) {
+		return;
+	}
+
+	if (typeof to[key] === 'undefined' || !isObj(val)) {
```

Comment:
Damn! [ECMAScript2015](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-object.assign) says we must throw a `TypeError` (`ToObject(null)`).
And the same applies to `{foo: undefined}` :disappointed: